kube/service.go

```diff
@@ -98,28 +67,48 @@ func NewClusterIPService(role *model.Role, headless bool, public bool, settings
 		}
 
 		for _, portInfoEntry := range portInfos {
-			svcPort := apiv1.ServicePort{
-				Name:     portInfoEntry.name,
-				Port:     portInfoEntry.port,
-				Protocol: protocol,
-			}
+			port := helm.NewStrMapping(
+				"name", portInfoEntry.name,
+				"port", strconv.Itoa(portInfoEntry.port),
+				"protocol", strings.ToUpper(portDef.Protocol),
+			)
 			if !headless {
-				svcPort.TargetPort = intstr.FromString(portInfoEntry.name)
-			}
-			service.Spec.Ports = append(service.Spec.Ports, svcPort)
-		}
-		if public {
-			if settings.CreateHelmChart {
-				service.Spec.ExternalIPs = []string{"{{ .Values.kube.external_ip | quote }}"}
-			} else {
-				service.Spec.ExternalIPs = []string{"192.168.77.77"}
+				port.AddStr("targetPort", portInfoEntry.name)
 			}
+			ports = append(ports, port)
 		}
 	}
-	if len(service.Spec.Ports) == 0 {
+	if len(ports) == 0 {
 		// Kubernetes refuses to create services with no ports, so we should
 		// not return anything at all in this case
 		return nil, nil
 	}
+
+	spec := helm.NewEmptyMapping()
+	spec.Add("selector", helm.NewStrMapping("skiff-role-name", role.Name))
```

Comment:
Hmm, we dropped the `RoleNameLabel` constant? And I thought we had newer helper methods for these.

Reply:
It got un-dropped in https://github.com/SUSE/fissile/commit/8e841ab22bd42d94978f4b850e40f0d0861c7321